src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private ExecutorService executor = Executors.newFixedThreadPool(5);
+    private List<EventListener> listenersList = new CopyOnWriteArrayList<>();
 
+    public void registerListener(EventListener listener) {
+        listenersList.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listenersList.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        listenersList.forEach(listener -> executor.submit(() -> listener.onEvent(event)));
     }
 
     public void shutdown() {
-
+        executor.shutdown();
```

Comment:
When shutting down the executor, it's recommended to first call `shutdown` to stop accepting new tasks, and then call `awaitTermination` with a timeout to ensure that all tasks have completed before continuing with the shutdown process.